### ISSUE TITLE
[Help wanted] Surface Wind Application of BCCAQ

### ISSUE THREAD

**Issue by nannau** [pacificclimate/ClimDown#18]
Hello PCIC!

I'd like to use BCCAQ as a baseline for a downscaling method I'm developing. As the issue title suggests, I'd like to apply it to wind components of synchronized (in time) climate models. 

**The models:**

ERA Interim (coarse resolution grids): 16x16 region with 6 hourly U and V wind components from 2000 to 2013.
WRF (fine reslution grids): 128x128 region identical to ERA Interim with 6 hourly U and V wind components from 2000 to 2013.

**My goal:**

My intention is basically to just downscale and bias correct a "projection period" on the univariate statistics of each wind component. Since WRF is driven by ERA-Interim, there _shouldn't_ be much a model to model bias between them, but if there is, I'd like to remove it.

As I currently understand it, BCCAQ requires a historical period for calibration and a projection period. My calibration period exists in a "train" subset of years from 2000 to 2013 (namely, 2001 - 2005, 2007 - 2009, and 2011 - 2013). The validation (or projection period) is the compliment subset of the training set of the complete years: 2000, 2006, and 2010. I would like to bias correct and downscale this validation set from the coarse ERA Interim grid to the fine WRF grid for 2000, 2006 and 2010 using the years in the training set as a calibration. Would this smattering of projection years (not being a true future projection period) cause issues in the pipeline? If this is an issue, I could manufacture a fake projection period that works. The times themselves do not really matter. 

Also, in the ClimDown documentation and BCCAQv2 methods document acquired last year when I was at PCIC, typically the climate variables mentioned only include precipitation and temperature - both of which respectively apply a ratio delta or an additive delta. I'm not actually sure which is more appropriate for wind fields, but I think zero velocity fields would cause issue, so an additive delta is preferred. I'm unsure if this method can be supplied as an option when running the pipeline, or if it automatically selects it from the variable name, in which case wind would cause an issue.

In summary, with the above use case mentioned, does it sound like ClimDown can be adopted to use with these wind fields?

Thanks for your time!

**Comment by jameshiebert** [comment]
> I would like to bias correct and downscale this validation set from the coarse ERA Interim grid to the fine WRF grid for 2000, 2006 and 2010 using the years in the training set as a calibration.

From a *software* standpoint, I believe that ClimDown just uses whatever you provide it. So make sure that if you only want to include those years, that only those years are present in the data you provide. I.e. don't give it a timeseries that has the intervening years present and set to NA.

> Again, the software itself is able to do whatever variables you give it, but it does have a number of per-variable config options, whose defaults do not include defaults for wind (or other variables). But have a close read of the `options` section in [the ClimDown manual](https://cran.r-project.org/web/packages/ClimDown/ClimDown.pdf) and you should be able to figure out how to redefine the defaults to your purposes. You'll probably do something like:

```R
options(ratio=list(wind=FALSE), seasonal=list(wind=...))
```

Scientifically, I can't speak to either of those things, but I'm sure you have resources at your disposal who can :smile:

**Comment by nannau** [comment]
Okay, thanks. Good to know that this might be possible. I have read the manual, but your example above clears up what the ratio and seaonal options actually do.

Another question I have is how it deals with sub-daily variables - like my 6 hourly wind. I didn't see any examples of this, but in the documentation I've read (and the title), there are only references to daily data. Could this be a problem?

**Comment by jameshiebert** [comment]
> clears up what the ratio and seasonal options actually do

Yeah and for other options (i.e. all of the options that say "not recommended to change"), I'm sure Alex, Arelia or Stephen can help understand whether and how to set them.

> Could this be a problem?

My recollection is that ClimDown only operates on a per-timestep basis. I.e. it doesn't do any aggregation/disaggregation along the time axis. It certainly does lots of rearranging of timesteps in the Climate Analogues and `rerank`. So again you'd have to check with the Alex, Arelia, or Stephen as to whether the assumptions that go into those processes are valid at a sub-daily scale. I can't speak to that.

**Comment by nannau** [comment]
Yeah, Alex actually just suggested that I skirt this by synthesizing a fake time series with each 6 hourly fields occuring daily. 

I'll go ahead and close this and maybe open a more technical issue if I run into problems. Thanks again, James!
